src/lib/schedule.c

```diff
@@ -309,6 +309,9 @@ void schedule_task_complete(struct task *task)
 
 	tracev_pipe("com");
 
+	if (task->state == TASK_STATE_COMPLETED)
```

Comment:
Best to remove list_item_del() so that we remove the task list only in one place.

Reply:
@lrgirdwo  yes, it is only deleted in this function. But this function is called by pipeline_schedule_cancel or _irq_task

Reply:
So the list_item_del() should be kept here.